lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -244,20 +243,20 @@ def associate_ip_and_device(ec2, address, private_ip_address, device_id, allow_r
     if not check_mode:
         if is_instance:
             if address.domain == "vpc":
-                res = ec2.associate_address(device_id,
-                                            allocation_id=address.allocation_id,
-                                            private_ip_address=private_ip_address,
-                                            allow_reassociation=allow_reassociation)
+                res = ec2.associate_address(InstanceId=device_id,
```

Comment:
Could you please add `AWSRetry` to the API calls in this module?  This helps a lot with intermitten API failures than can crop up in CI or for users.

Reply:
@jillr added